### PR TITLE
CC-25819 Integration Guide for backoffice login brute force security issue

### PR DESCRIPTION
**Pull request by artem-png**
## PR Description

Security Fix: https://spryker.atlassian.net/browse/CC-16903
IG Jira ticket: https://spryker.atlassian.net/browse/CC-25819

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: _includes/pbc/all/install-features/202304.0/install-the-spryker-core-back-office-feature.md [hunk 1]
```diff
@@ -71,33 +73,40 @@ console transfer:generate
...
+| OauthAuthenticationLinkTransfer      | class     | created | src/Generated/Shared/Transfer/OauthAuthenticationLinkTransfer              |
+| ResourceOwnerTransfer                | class     | created | src/Generated/Shared/Transfer/ResourceOwnerTransfer                        |
+| ResourceOwnerRequestTransfer         | class     | created | src/Generated/Shared/Transfer/ResourceOwnerRequestTransfer                 |
+| ResourceOwnerResponseTransfer        | class     | created | src/Generated/Shared/Transfer/ResourceOwnerResponseTransfer                |
+| SecurityBlockerConfigurationSettings | class     | created | src/Generated/Shared/Transfer/SecurityBlockerConfigurationSettingsTransfer |
+| SecurityCheckAuthResponse            | class     | created | src/Generated/Shared/Transfer/SecurityCheckAuthResponseTransfer            |
+| SecurityCheckAuthContext             | class     | created | src/Generated/Shared/Transfer/SecurityCheckAuthContextTransfer             |
+| Locale                               | class     | created | src/Generated/Shared/Transfer/LocaleTransfer                               |
```
Comment (DmitryLymarenko) [inline _includes/pbc/all/install-features/202304.0/install-the-spryker-core-back-office-feature.md:93]: `Transfer` postfix is missing

File: _includes/pbc/all/install-features/202304.0/install-the-spryker-core-back-office-feature.md [hunk 2]
```diff
@@ -285,6 +302,54 @@ class SecurityDependencyProvider extends SprykerSecurityDependencyProvider
...
+
+```php
+<?php
+
+namespace Pyz\Client\SecurityBlocker;
+
+use Spryker\Client\SecurityBlocker\SecurityBlockerDependencyProvider as SprykerSecurityBlockerDependencyProviderAlias;
+use Spryker\Client\SecurityBlockerBackoffice\Plugin\SecurityBlocker\BackofficeUserSecurityBlockerConfigurationSettingsExpanderPlugin;
+
+class SecurityBlockerDependencyProvider extends SprykerSecurityBlockerDependencyProviderAlias
```
Comment (DmitryLymarenko) [inline _includes/pbc/all/install-features/202304.0/install-the-spryker-core-back-office-feature.md:315]:
```suggestion
class SecurityBlockerDependencyProvider extends SprykerSecurityBlockerDependencyProvider
```

Comment (DmitryLymarenko) [reply _includes/pbc/all/install-features/202304.0/install-the-spryker-core-back-office-feature.md:315]: same for use statement

File: _includes/pbc/all/install-features/202304.0/install-the-spryker-core-feature.md [hunk 3]
```diff
@@ -40,6 +40,8 @@ Make sure that the following modules have been installed:
 | StorageRedis | vendor/spryker/storage-redis |
 | SecuritySystemUser | vendor/spryker/security-system-user |
 | SecurityBlocker | vendor/spryker/security-blocker |
+| SecurityBlockerExtension | vendor/spryker/security-blocker-extension |
+| SecurityBlockerStorefrontCustomer | vendor/spryker/security-storefront-customer |
```
Comment (DmitryLymarenko) [inline _includes/pbc/all/install-features/202304.0/install-the-spryker-core-feature.md:44]: name of the module and directory are different

File: _includes/pbc/all/install-features/202304.0/install-the-spryker-core-feature.md [hunk 4]
```diff
@@ -473,6 +468,31 @@ class StorageDependencyProvider extends SprykerStorageDependencyProvider
...
+
+```php
+<?php
+
+namespace Pyz\Client\SecurityBlocker;
+
+use Spryker\Client\SecurityBlocker\SecurityBlockerDependencyProvider as SprykerSecurityBlockerDependencyProviderAlias;
+use Spryker\Client\SecurityBlockerStorefrontCustomer\Plugin\SecurityBlocker\CustomerSecurityBlockerConfigurationSettingsExpanderPlugin;
+
+class SecurityBlockerDependencyProvider extends SprykerSecurityBlockerDependencyProviderAlias
```
Comment (DmitryLymarenko) [inline _includes/pbc/all/install-features/202304.0/install-the-spryker-core-feature.md:481]:
```suggestion
class SecurityBlockerDependencyProvider extends SprykerSecurityBlockerDependencyProvider
```

same for use statement

File: docs/marketplace/dev/feature-integration-guides/202304.0/marketplace-merchant-portal-core-feature-integration.md [hunk 5]
```diff
@@ -0,0 +1,1398 @@
...
+```
+
+**src/Pyz/Client/SecurityBlocker/SecurityBlockerDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Client\SecurityBlocker;
+
+use Spryker\Client\SecurityBlocker\SecurityBlockerDependencyProvider as SprykerSecurityBlockerDependencyProviderAlias;
```
Comment (DmitryLymarenko) [inline docs/marketplace/dev/feature-integration-guides/202304.0/marketplace-merchant-portal-core-feature-integration.md:212]:
```suggestion
use Spryker\Client\SecurityBlocker\SecurityBlockerDependencyProvider as SprykerSecurityBlockerDependencyProvider;
```

File: docs/marketplace/dev/feature-integration-guides/202304.0/marketplace-merchant-portal-core-feature-integration.md [hunk 6]
```diff
@@ -0,0 +1,1398 @@
...
+use Spryker\Client\SecurityBlockerMerchantPortal\Plugin\SecurityBlocker\MerchantPortalUserSecurityBlockerConfigurationSettingsExpanderPlugin;
+
+class SecurityBlockerDependencyProvider extends SprykerSecurityBlockerDependencyProviderAlias
```
Comment (DmitryLymarenko) [inline docs/marketplace/dev/feature-integration-guides/202304.0/marketplace-merchant-portal-core-feature-integration.md:215]:
```suggestion
class SecurityBlockerDependencyProvider extends SprykerSecurityBlockerDependencyProvider
```

**Review by DmitryLymarenko — CHANGES_REQUESTED** [review]

File: _includes/pbc/all/install-features/202304.0/install-the-agent-assist-feature.md [hunk 7]
```diff
@@ -0,0 +1,574 @@
...
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that when the login form for the agent is submitted, the URL it uses contains a locale code. For example, `/de/agent/login_check` is the default value for the agent.
+
+{% endinfo_block %}
+
+Make sure to use environment variables in `config-default.php`:
```
Comment (DmitryLymarenko) [inline _includes/pbc/all/install-features/202304.0/install-the-agent-assist-feature.md:202]: use the same wording for similar cases

for core feature you have `Add environment configuration for customer security:` and here you have `Make sure to use environment variables in config-default.php:`

File: _includes/pbc/all/install-features/202304.0/install-the-agent-assist-feature.md [hunk 8]
```diff
@@ -199,6 +199,20 @@ Make sure that when the login form for the agent is submitted, the URL it uses c
...
+
+**config/Shared/config_default.php**
+
+```php
+<?php
+// other code
+
+// >>> Security Blocker Storefront Agent
+$config[SecurityBlockerStorefrontAgentConstants::AGENT_BLOCK_FOR_SECONDS] = 360;
```
Comment (DmitryLymarenko) [inline _includes/pbc/all/install-features/202304.0/install-the-agent-assist-feature.md:211]: config constants are present at respective table for `SecurityBlockerBackofficeConstants` case, but these tables are missing for other cases

File: _includes/pbc/all/install-features/202304.0/install-the-agent-assist-feature.md (hunk 7 above)
Comment (artem-png) [reply _includes/pbc/all/install-features/202304.0/install-the-agent-assist-feature.md]: Adjusted everywhere and made it almost the same, MP core. MP core has different structure

**Review by DmitryLymarenko — DISMISSED** [review]

**Review by KarinOf — APPROVED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**